processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -301,6 +298,9 @@ public static String convertObjectToString(@Nullable Object valObj)
   {
     if (valObj == null) {
       return null;
+    } else if (valObj instanceof Object[]) {
+      // TODO(laksh): Get this change vetted
+      return Arrays.toString((Object[]) valObj);
```

Comment:
when does this happen?

Reply:
It happened in one of the funny cases that you alluded to before, where we try to compare arrays as strings. Since I wasn't able to find the test case directly, I have pushed a change without this method to see where it fails. Before the patch, when we used `ComparableStringArrays` and `ComparableList,` the `toString` was implicitly returning the required representation, however with the change, it started returning `Object@...`, which doesn't sit well with the test.

Reply:
Tests like the following require this change:
`ClientQuerySegmentWalkerTest#testGroupByOnArraysLongsAsString`
`ClientQuerySegmentWalkerTest#testGroupByOnArraysStrings`
`ClientQuerySegmentWalkerTest#testGroupByOnArraysStringsasString`

Reply:
i feel like we could probably consider just make this an error case too, maybe also worth a dev list thread along with dropping support for funny string comparators on non-string types. i'll try to start a dev list thread soon